<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1392</width>
    <height>887</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Python BDIWS - Alpha 0.1</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="2,3">
    <item>
     <layout class="QVBoxLayout" name="LeftSection" stretch="2,2,6,4,100">
      <item>
       <widget class="QGroupBox" name="GroupTypeManipulation">
        <property name="title">
         <string>水印操作方式</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QRadioButton" name="TypeAddWatermark">
             <property name="text">
              <string>添加水印</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">buttonGroupM</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="TypeExtractWatermark">
             <property name="text">
              <string>提取水印</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">buttonGroupM</string>
             </attribute>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="GroupTypeWatermark">
        <property name="title">
         <string>水印类型选择</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QRadioButton" name="TypeMonochrome">
             <property name="text">
              <string>黑白图像</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">buttonGroupT</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="TypeString">
             <property name="text">
              <string>字符串</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">buttonGroupT</string>
             </attribute>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="GroupTargetOriginal">
        <property name="title">
         <string>待操作目标</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5" stretch="1,1,1">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="10,1,20,5">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>待处理的图像路径</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QTextBrowser" name="LabelPathOriginal">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ButtonPathOriginal">
             <property name="text">
              <string>浏览</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="10,1,20,5">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>待添加的水印路径</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QTextBrowser" name="LabelPathMonochrome">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ButtonPathMonochrome">
             <property name="text">
              <string>浏览</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="10,1,20,5">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>待添加的字符串水印</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="LabelStringOriginal">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QPushButton" name="ButtonStringClear">
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="ButtonStringPaste">
               <property name="text">
                <string>粘贴</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="GroupTargetFinished">
        <property name="title">
         <string>待导出目标</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0,1,1">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_18" stretch="10,1,20,5">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>加密图像导出路径</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QTextBrowser" name="LabelDirImgOut">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ButtonDirImg">
             <property name="text">
              <string>浏览</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="10,1,20,5">
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>图像水印提取路径</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QTextBrowser" name="LabelDirMonochrome">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ButtonDirMonochrome">
             <property name="text">
              <string>浏览</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="10,1,20,5">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>提取得到的字符串</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QTextBrowser" name="LabelStringExtract">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ButtonStringCopy">
             <property name="text">
              <string>复制</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>信息提示区</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_15">
         <item>
          <widget class="QTextBrowser" name="InfoScreen"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout" rowstretch="1,0" columnstretch="1,0">
      <item row="0" column="1">
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>水印</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="QLabel" name="GraphWatermark">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>350</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>5000</width>
             <height>5000</height>
            </size>
           </property>
           <property name="text">
            <string>[N/A]</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>添加水印后的图像</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_13">
         <item>
          <widget class="QLabel" name="GraphProcessed">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>400</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>5000</width>
             <height>5000</height>
            </size>
           </property>
           <property name="text">
            <string>[N/A]</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>原图像</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_12">
         <item>
          <widget class="QLabel" name="GraphOriginal">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>350</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>5000</width>
             <height>5000</height>
            </size>
           </property>
           <property name="text">
            <string>[N/A]</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_17">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="topMargin">
           <number>80</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <item>
           <widget class="QPlainTextEdit" name="LabelPasswordWm">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>45</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="placeholderText">
             <string>请输入水印加密/解密的密码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_20">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <item>
           <widget class="QPlainTextEdit" name="LabelPasswordImg">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>45</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="placeholderText">
             <string>请输入图像加密/解密的密码</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_23" stretch="10,1,10">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <item>
           <widget class="QPlainTextEdit" name="LabelWmWidth">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>45</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="placeholderText">
             <string>请输入被提取水印的宽</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_7">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="LabelWmHeight">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>45</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="placeholderText">
             <string>请输入被提取水印的高</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_19">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <item>
           <widget class="QPlainTextEdit" name="LabelAntiAtt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>408</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>45</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="placeholderText">
             <string>请输入鲁棒性等级，留空则为默认（36）</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_16">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>100</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButtonClearPara">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>清除所有参数</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonExecute">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>执行并保存</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1392</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="MenuHelp">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="ActionAbout"/>
   </widget>
   <addaction name="MenuHelp"/>
  </widget>
  <action name="ActionAbout">
   <property name="text">
    <string>关于 Python BDIWS</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroupT"/>
  <buttongroup name="buttonGroupM"/>
 </buttongroups>
</ui>
